crates/agglayer-aggregator-notifier/src/settlement_client/rpc.rs

```diff
@@ -0,0 +1,385 @@
+            }
+        };
+
+        info!(
+            "Initializing the settlement on L1 with public inputs: {}",
+            output.display_to_hex()
+        );
+
+        // Record the settlement parameters for tracing
+        let settlement_params = format!(
+            "origin_network: {}, l1_info_tree_leaf_count: {}, new_local_exit_root: 0x{}, \
+             new_pessimistic_root: 0x{}, proof_length: {}, custom_chain_data_length: {}",
+            output.origin_network.to_u32(),
+            l1_info_tree_leaf_count,
+            hex::encode(output.new_local_exit_root),
+            hex::encode(output.new_pessimistic_root),
+            proof_with_selector.len(),
+            certificate.custom_chain_data.len()
+        );
+        tracing::Span::current().record("settlement_params", &settlement_params);
+
+        // Step 6: Call the contract settlement function and get the pending transaction
+        let pending_tx = match self
+            .l1_rpc
+            .verify_pessimistic_trusted_aggregator(
+                output.origin_network.to_u32(),
+                l1_info_tree_leaf_count,
+                *output.new_local_exit_root.as_ref(),
+                *output.new_pessimistic_root,
+                proof_with_selector.into(),
+                certificate.custom_chain_data.into(),
+            )
+            .await
+        {
+            Ok(pending_tx) => {
+                info!("Certificate settlement transaction submitted");
+                pending_tx
+            }
+            Err(error) => {
+                let error_str = RollupManagerRpc::decode_contract_revert(&error)
+                    .unwrap_or_else(|| error.to_string());
+
+                error!(
+                    error_code = %error,
+                    error = error_str,
+                    "Failed to settle certificate"
+                );
+
+                return Err(Error::SettlementError {
+                    certificate_id,
+                    error: error_str,
+                });
+            }
+        };
+
+        // Get the transaction hash from the pending transaction
+        let tx_hash = *pending_tx.tx_hash();
+        info!("Settlement transaction hash: {}", tx_hash);
+
+        Ok(SettlementTxHash::from(tx_hash))
+    }
+
+    #[tracing::instrument(skip(self))]
+    async fn wait_for_settlement(
+        &self,
+        settlement_tx_hash: SettlementTxHash,
+        certificate_id: CertificateId,
+    ) -> Result<(EpochNumber, CertificateIndex), Error> {
+        info!(%settlement_tx_hash, "Waiting for settlement of tx {settlement_tx_hash}");
+
+        // Step 1: Wait for transaction receipt with retries
+        let receipt = self
+            .wait_for_transaction_receipt(settlement_tx_hash, certificate_id)
+            .await?;
+
+        #[cfg(feature = "testutils")]
+        // Apply fail points if they are active for integration testing
+        testutils::inject_settle_certificate_fail_points(certificate_id)?;
```

Comment:
Tiny nit: I'd find it more natural to swap the order of the comment and the attribute.
```suggestion
        // Apply fail points if they are active for integration testing
        #[cfg(feature = "testutils")]
        testutils::inject_settle_certificate_fail_points(certificate_id)?;
```